crates/streamer/src/ascii_streamer.rs

```diff
@@ -0,0 +1,68 @@
+use std::future::Future;
+
+use tracing::instrument;
+
+use crate::Signal;
+
+use tokio::io::{AsyncBufReadExt, BufReader};
+
+pub trait AsciiStreamSocket {
+    fn send(&mut self, value: String) -> impl Future<Output = anyhow::Result<()>> + Send;
+    fn close(self);
+}
```

Comment:
question: Do you really need a new trait, or you could work with AsyncWrite? There is an adapter to get map a websocket into a AsyncRead + AsyncWrite object: https://github.com/Devolutions/devolutions-gateway/blob/0016f78df77a08bbf707e9aa0e248c171756a3fc/devolutions-gateway/src/ws.rs#L5

Reply:
The client need text specifically, if we send bytes it won't work.